### PR TITLE
Auth0 express api

### PR DESCRIPTION
**Pull request by LeiaQuesada**
https://trello.com/c/AkOgihzn/43-secure-express-server-with-auth0

this branch is for adding auth0 to the backend. (Client side has been secured by wrapping Auth0 around user login authentication)

I've imported and called the Auth0 middleware for APIs and it successfully passes the jwt (token) from express routes. 

Problem: React route to express path is not working. Kicks back error about misuse of importing auth0(). I'll point to it in an apiClient.js PR comment

### CODE REVIEW

File: app/src/apiClient.js [hunk 1]
```diff
@@ -13,3 +15,16 @@ export const addTask = async (name) => {
   });
   return response.json();
 };
+
+export const testAuth = async () => {
+  //FIXME: get JWT (need to resolve React-related error)
```
Comment (m3ia) [inline app/src/apiClient.js:20]: Do you still need these?

File: app/src/index.js [hunk 2]
```diff
@@ -12,9 +12,9 @@ ReactDOM.render(
     <>
       <h1>MyGarden</h1>
       <Auth0Provider
-        domain="dev-wbfyt2d4.us.auth0.com"
-        clientId="qAVrIAowOt9AM2e1pWSfWDLDgBJfAqX6"
-        redirectUri={window.location.origin}
+        domain="dev-wbfyt2d4.us.auth0.com" // from Single Page app
+        clientId="qAVrIAowOt9AM2e1pWSfWDLDgBJfAqX6" // from SPA
+        audience="https://dev-wbfyt2d4.us.auth0.com/api/v2/" // from API
```
Comment (m3ia) [inline app/src/index.js:17]: Nice notes

File: server/server.mjs [hunk 3]
```diff
@@ -1,10 +1,34 @@
...
 import mime from "mime-types";
 
 import * as db from "./db.mjs";
 
 const app = express();
 const port = process.env.PORT || 4000;
+// TODO: CONSIDER checking scopes
+// import jwtAuthz from "express-jwt-authz";
+
+// Authorization middleware. When used, the
```
Comment (m3ia) [inline server/server.mjs:13]: Optional: could change 3-line comment to follow JS standards /* */

**Review by m3ia — APPROVED** [review]
Auth0 works on my end (I was able to sign in using my Google account!). Syntax looks good.

File: app/src/apiClient.js (hunk 1 above)
Comment (gsong) [reply app/src/apiClient.js:20]: @LeiaQuesada We can work on how to get the token to all you API client calls in a future session.

**Review by gsong — APPROVED** [review]

**Pull request merged**